src/cephadm/cephadm

```diff
@@ -486,12 +461,129 @@ class CephIscsi(object):
+                 config_json: Dict, image: str) -> None:
+        self.fsid = fsid
+        self.daemon_id = daemon_id
+        self.image = image
+
+        # config-json options
+        self.entrypoint = dict_get(config_json, 'entrypoint')
+        self.uid = dict_get(config_json, 'uid', 65534)  # nobody
+        self.gid = dict_get(config_json, 'gid', 65534)  # nobody
+        self.args = dict_get(config_json, 'args', [])
+        self.volume_mounts = dict_get(config_json, 'volume_mounts', {})
+        self.container_args = dict_get(config_json, 'container_args', [])
+        self.envs = dict_get(config_json, 'envs', [])
+        self.privileged = dict_get(config_json, 'privileged', False)
+        self.ptrace = dict_get(config_json, 'ptrace', False)
+        self.bind_mounts = dict_get(config_json, 'bind_mounts', [])
+        self.ports = dict_get(config_json, 'ports', [])
+        self.dirs = dict_get(config_json, 'dirs', [])
+        self.files = dict_get(config_json, 'files', {})
+
+    @classmethod
+    def init(cls, fsid: str, daemon_id: Union[int, str]) -> 'CustomContainer':
+        return cls(fsid, daemon_id, get_parm(args.config_json), args.image)
+
+    def create_daemon_dirs(self, data_dir: str, uid: int, gid: int) -> None:
+        """
+        Create dirs/files below the container data directory.
+        """
+        logger.info('Creating custom container configuration dirs/files in %s...' % data_dir)
+
+        if not os.path.isdir(data_dir):
+            raise OSError('data_dir is not a directory: %s' % data_dir)
+
+        for dir_path in self.dirs:
+            logger.info('Creating directory: %s' % dir_path)
+            dir_path = os.path.join(data_dir, dir_path.strip('/'))
+            makedirs(dir_path, uid, gid, 0o755)
+
+        for file_path in self.files:
+            logger.info('Creating file: %s' % file_path)
+            content = dict_get_join(self.files, file_path)
+            file_path = os.path.join(data_dir, file_path.strip('/'))
+            with open(file_path, 'w') as f:
+                os.fchown(f.fileno(), uid, gid)
+                os.fchmod(f.fileno(), 0o600)
+                f.write(content)
+
+    def get_daemon_args(self) -> List[str]:
+        return self.args
+
+    def get_container_args(self) -> List[str]:
+        return self.container_args
+
+    def get_container_envs(self) -> List[str]:
+        return self.envs
+
+    def get_container_mounts(self, data_dir: str) -> Dict[str, str]:
+        mounts = {}
+        for key, value in self.volume_mounts.items():
+            key = re.sub(r'{{\s*data_dir\s*}}', data_dir, key)
+            mounts[key] = value
+        return mounts
+
+    def get_container_binds(self, data_dir: str) -> List[List[str]]:
+        binds = self.bind_mounts.copy()
+        for bind in binds:
+            for index, value in enumerate(bind):
+                bind[index] = re.sub(r'{{\s*data_dir\s*}}', data_dir, value)
```

Comment:
what do you think of instead of:


```yaml
volume_mounts:
    "{{ data_dir }}/DATA_DIR": /data 
```
we treat relative dirs as relative to `data_dir`? 
```yaml
volume_mounts:
    "DATA_DIR": /data # relative to data_dir
    "/my/absolute/path": /path # absolute
```